extensions/cdc-core/src/main/java/com/hazelcast/jet/cdc/ChangeEventValue.java

```diff
@@ -0,0 +1,94 @@
+
+import com.hazelcast.jet.annotation.EvolvingApi;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Value (body) portion of a {@link ChangeEvent}, describes the actual
+ * change that has been done to the affected database record or
+ * document.
+ *
+ * @since 4.1
+ */
+@EvolvingApi
+public interface ChangeEventValue extends ChangeEventElement {
+
+    /**
+     * Specifies the moment in time when the event happened. This
+     * timestamp is "real" in the sense that it comes from the database
+     * change-log, so it's not "processing time", it's not the moment
+     * when the event was observed by our system. Keep in mind though
+     * that not all events come from the change-log. The change-log goes
+     * back in time only to a limited extent, all older events are parts
+     * of a database snapshot constructed when we start monitoring the
+     * database and their timestamps are accordingly artificial.
+     * Identifying snapshot events is possible most of the time, because
+     * their operation will be {@link Operation#SYNC} instead of
+     * {@link Operation#INSERT} (one notable exception being MySQL).
+     *
+     * @throws ParsingException if no parsable timestamp field present
+     */
+    long timestamp() throws ParsingException;
+
+    /**
+     * Specifies the type of change being described (insertion, delete or
+     * update). Only some special events, like heartbeats don't have an
+     * operation value.
+     *
+     * @return {@link Operation#UNSPECIFIED} if this {@code ChangeEventValue}
+     * doesn't have an operation field or appropriate {@link Operation}
+     * that matches what's found in the operation field
+     * @throws ParsingException if there is an operation field, but it's
+     *                          value is not among the handled ones.
+     */
+    @Nonnull
+    Operation operation() throws ParsingException;
+
+    /**
+     * Describes how the database record or document looked like BEFORE
+     * applying the change event. Not provided for MongoDB updates.
+     *
+     * @throws ParsingException if this {@code ChangeEventValue} doesn't
+     *                          have a 'before' sub-element
+     */
+    @Nonnull
+    ChangeEventElement before() throws ParsingException;
+
+    /**
+     * Describes how the database record or document looks like AFTER
+     * the change event has been applied. Not provided for MongoDB updates.
+     *
+     * @throws ParsingException if this {@code ChangeEventValue} doesn't
+     *                          have an 'after' sub-element
+     */
+    @Nonnull
+    ChangeEventElement after() throws ParsingException;
+
+    /**
+     * Describes the change being done by the event. Only used by
+     * MongoDB updates.
+     *
+     * @throws ParsingException if this {@code ChangeEventValue} doesn't
+     *                          have an 'patch' sub-element
+     */
+    @Nonnull
+    ChangeEventElement change() throws ParsingException;
```

Comment:
Do we really want to make this MongoDb specific concern first class citizen of the API ?

Reply:
Assuming I remove it, what will I do when I do want to include Mongo later? Just add the method then, would that be ok?

Reply:
No, maybe from scratch mongo source can emit a different event which contains `patch` and `filter` fields in case of an update event.